src/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -262,15 +309,86 @@ internal IPAddress(int newAddress)
         /// </devdoc>
         public static bool TryParse(string ipString, out IPAddress address)
         {
+            if (ipString == null)
+            {
+                address = null;
+                return false;
+            }
+
             address = IPAddressParser.Parse(ipString, true);
             return (address != null);
         }
 
+        public static bool TryParse(ReadOnlySpan<char> ipSpan, out IPAddress address)
+        {
+            if (ipSpan.IsEmpty)
+            {
+                address = null;
+                return false;
+            }
+
+            address = IPAddressParser.Parse(ipSpan, true);
```

Comment:
Nit: I know the existing code doesn't have this, but it'd be nice to name the Boolean argument at the call site; otherwise it's difficult to know while reading the code what it means. (Also the parens around the expression below are unnecessary... again, I realize you were just copying the existing style.)